src/imgix-core-js.js

```diff
@@ -124,6 +124,74 @@
       }
     };
 
+    ImgixClient.prototype.buildSrcSet = function (path, params) {
+      var width = params ? params['w'] : undefined;
+      var height = params ? params['h'] : undefined;
+      var aspectRatio = params ? params['ar'] : undefined;
+
+      // determines if an aspect ratio value is in the correct format 'w:h'
+      var isValidFormat = function() {
+        return /^\d+(\.\d+)?:\d+(\.\d+)?$/.test(aspectRatio);
+      }();
+
+      if (aspectRatio && !isValidFormat) {
+        throw new Error('The \'ar\' parameter key must follow the format w:h');
+      }
+
+      var fixedWidth = ((width && height) || (width && aspectRatio) || (height && aspectRatio)) ? true : false;
+
+      if (fixedWidth) {
+        return this._buildDPRSrcSet(path, params);
+      }
+      else {
+        return this._buildSrcSetPairs(path, params);
+      }
+    };
+
+    ImgixClient.prototype._buildSrcSetPairs = function(path, params) {
+      var srcset = '';
+      var targetWidths = function() {
+        var resolutions = [];
+        var prev = 100;
+        var INCREMENT_PERCENTAGE = 8;
+        var MAX_SIZE = 8192;
+      
+        var ensureEven = function(n){
+          return 2 * Math.round(n / 2);
+        };
+      
+        while (prev <= MAX_SIZE) {
+          resolutions.push(ensureEven(prev));
+          prev *= 1 + (INCREMENT_PERCENTAGE / 100) * 2;
+        }
+      
+        resolutions.push(MAX_SIZE);
+        return resolutions;
+      }();
+
+      for(var i = 0; i < targetWidths.length; i++) {
+        currentWidth = targetWidths[i];
```

Comment:
This variables aren't declared in this function, so they're gonna end up on the global scope. We should do a `var currentWidth, currentParams;` at the top of the function (or right before this loop, since we don't seem to care about hoisting variable declarations anywhere else).